### ISSUE TITLE
HTTP/3 with OpenSSL back end leaves connections hanging on IPv6

### ISSUE THREAD

**Issue by Fullmetal5** [curl/curl#14022]
### I did this

Attempting to make a connection with HTTP/3 using the OpenSSL back end and using IPv6 I noticed in Wireshark that libcurl/openssl just stop replying to packets as soon as the server sends its reply and the application attempts to exit. This leaves the server re-transmitting the reply having never gotten an ACK yet alone a CC. I'm not sure if this is a libcurl bug or a OpenSSL bug as I haven't tested yet using different back ends.

Example command to run with Wireshark running:
`SSLKEYLOGFILE=/tmp/keys.txt curl -6 -I --header 'if-modified-since: Wed, 26 Jun 2024 05:21:08 GMT' 'https://example.com/' -o /dev/null -s --http3-only`

Loading the keys.txt and taking a look I can see in my capture that the server response with '304 Not Modified' repeatedly but curl never sends an ACK or CC leaving the server to just re-transmit till it too hangs up.

### I expected the following

For there to at least be an attempt at a graceful connection shutdown rather than leaving the server hanging.

### curl/libcurl version

Testing on curl 8.8.0 (Arch Linux packaged) and master (c1845dc0e2329685059f3533dc296c182a47a1aa).
Also tested with icing's "graceful connection shutdown" patch applied since the name seemed relevant to the issue but no change.

### operating system

`Linux 6.9.6-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 21 Jun 2024 19:49:19 +0000 x86_64 GNU/Linux`

**Comment by icing** [comment]
Thanks for the report. Where our OpenSSL QUIC shutdown was supposed to go, there was a `TODO` there instead.

I added support in #14027. Hope that works for you as well.

**Comment by Fullmetal5** [comment]
Yup, that has indeed fixed the issue, thank you for your hard work on the QUIC and HTTP/3 support.
